code/controllers/subsystem/ticker.dm

```diff
@@ -120,43 +130,47 @@ SUBSYSTEM_DEF(ticker)
 
 /datum/controller/subsystem/ticker/proc/setup()
 	cultdat = setupcult()
-	//Create and announce mode
-	if(GLOB.master_mode=="secret")
-		src.hide_mode = 1
+
+	// Create and announce mode
+	if(GLOB.master_mode == "secret")
+		hide_mode = TRUE
+
 	var/list/datum/game_mode/runnable_modes
-	if((GLOB.master_mode=="random") || (GLOB.master_mode=="secret"))
+
+	if(GLOB.master_mode == "random" || GLOB.master_mode == "secret")
 		runnable_modes = config.get_runnable_modes()
-		if(runnable_modes.len==0)
+		if(!length(runnable_modes))
 			to_chat(world, "<B>Unable to choose playable game mode.</B> Reverting to pre-game lobby.")
 			force_start = FALSE
 			current_state = GAME_STATE_PREGAME
 			Master.SetRunLevel(RUNLEVEL_LOBBY)
-			return 0
+			return FALSE
 		if(GLOB.secret_force_mode != "secret")
 			var/datum/game_mode/M = config.pick_mode(GLOB.secret_force_mode)
 			if(M.can_start())
-				src.mode = config.pick_mode(GLOB.secret_force_mode)
+				mode = config.pick_mode(GLOB.secret_force_mode)
 		SSjobs.ResetOccupations()
-		if(!src.mode)
-			src.mode = pickweight(runnable_modes)
-		if(src.mode)
-			var/mtype = src.mode.type
-			src.mode = new mtype
+		if(!mode)
+			mode = pickweight(runnable_modes)
+		if(mode)
+			var/mtype = mode.type
```

Comment:
```suggestion
		if(!mode)
			mode = pickweight(runnable_modes)
		else
			var/mtype = mode.type
```

Reply:
Its intentionally like this. A mode can be picked on like 154 then accepted on 155. If you change this to an `else`, like 156 will never trigger.